src/components/text/utils.ts

```diff
@@ -1,3 +1,37 @@
+import {Platform, TextStyle} from 'react-native';
+
 export const MAX_FONT_SCALE = 2;
 
 export const screenReaderPause = '\n';
+
+export function fontWeightToRobotoFamily(weight?: string) {
+  switch (weight) {
+    case '400':
+      return 'Roboto-Regular';
+    case '500':
+      return 'Roboto-Medium';
+    case '600':
+    case 'bold':
+      return 'Roboto-SemiBold';
+    default:
+      return 'Roboto-Regular';
+  }
+}
+
+export function getTextWeightStyleWithCustomAndroidHandling(
```

Comment:
I think this one can be shortened to just `getTextWeightStyle`, since we pass `androidSystemFont`, and that implicitly says that there is a custom handling for Android OS. The function declaration also have a check if the user has Android OS, so it feels kind of redundant.